pyproject.toml

```diff
@@ -39,6 +39,7 @@ dependencies = [
   "sqlalchemy>=1.3.0",
   "tqdm",
   "PyYAML",  # Only used in `optuna/cli.py`.
+  "typing_extensions; python_version<'3.8'",
```

Comment:
The `typing_extensions` package may be subject to version restrictions. See https://github.com/optuna/optuna/blob/ab958d42e44cad2e0517907278d55ca7432ea32c/pyproject.toml#L63.